src/QueryBuilderParser/JoinSupportingQueryBuilderParser.php

```diff
@@ -95,28 +95,51 @@ private function buildSubclauseQuery($query, $rule, $value, $condition)
 
         $not = array_key_exists('not_exists', $subclause) && $subclause['not_exists'];
 
-        // Create a where exists clause to join to the other table, and find results matching the criteria
-        $query = $query->whereExists(
-            /**
-             * @param Builder $query
-             */
-            function(Builder $query) use ($subclause) {
-
-                $q = $query->selectRaw(1)
-                    ->from($subclause['to_table'])
-                    ->whereRaw($subclause['to_table'].'.'.$subclause['to_col']
-                        .' = '
-                        .$subclause['from_table'].'.'.$subclause['from_col']);
-
-                if (array_key_exists('to_clause', $subclause)) {
-                    $q->where($subclause['to_clause']);
-                }
-
-                $this->buildSubclauseInnerQuery($subclause, $q);
-            },
-            $condition,
-            $not
-        );
+	    if ( $not ) {
+		    // Create a where exists clause to join to the other table, and find results matching the criteria
+		    $query = $query->whereExists(
+		    /**
+		     * @param Builder $query
+		     */
+			    function(Builder $query) use ($subclause, $not, $condition) {
+				    $q = $query->selectRaw(1)
+				               ->from($subclause['to_table'])
+				               ->whereRaw($subclause['to_table'].'.'.$subclause['to_col']
+				                          .' = '
+				                          .$subclause['from_table'].'.'.$subclause['from_col']);
+				    if ( ! isset( $subclause["through"] ) ) {
+					    if ( array_key_exists( 'to_clause', $subclause ) ) {
+						    $q->where( $subclause['to_clause'] );
+					    }
+					    $this->buildSubclauseInnerQuery( $subclause, $q );
+				    } else {
+					    $this->buildSubclauseThroughQuery( $subclause, $not, $condition, $q );
+				    }
+
+			    },
+			    $condition,
+			    $not
+		    );
+	    } else {
+		    // Create a join clause to join to the other table, and find results matching the criteria
+
+		    $query = $query->join( $subclause["to_table"], $subclause['to_table'] . '.' . $subclause['to_col'], '=', $subclause['from_table'] . '.' . $subclause['from_col'] );
```

Comment:
Please try to keep lines < 80 characters long. This line of code wraps in my browser.